src/main/java/org/jenkinsci/plugins/orgfolder/github/RepositoryGHEventSubscriber.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.orgfolder.github;
+
+import com.cloudbees.jenkins.GitHubRepositoryName;
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.security.ACL;
+import jenkins.branch.OrganizationFolder;
+import jenkins.scm.api.SCMSourceOwner;
+import jenkins.scm.api.SCMSourceOwners;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.github.extension.GHEventsSubscriber;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator;
+import org.kohsuke.github.GHEvent;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.Sets.immutableEnumSet;
+import static java.util.logging.Level.*;
+import static org.kohsuke.github.GHEvent.REPOSITORY;
+
+/**
+ * This subscriber manages {@link org.kohsuke.github.GHEvent} REPOSITORY.
+ */
+@Extension
+public class RepositoryGHEventSubscriber extends GHEventsSubscriber {
```

Comment:
When you get around implementing it on `scm-api`, please feel free to come back and remove it. At worst this provides extra folder computation, which is idempotent. I am keeping this until then, because this is a useful feature today.